schema/pgsql/schema.sql

```diff
@@ -2171,6 +2171,27 @@ COMMENT ON COLUMN sla_history_downtime.downtime_id IS 'downtime.id (may referenc
 COMMENT ON COLUMN sla_history_downtime.downtime_start IS 'start time of the downtime';
 COMMENT ON COLUMN sla_history_downtime.downtime_end IS 'end time of the downtime';
 
+CREATE TABLE sla_lifecycle (
+  id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  host_id bytea20 NOT NULL,
+  service_id bytea20 DEFAULT NULL,
+
+  -- These columns are nullable, but as we're using the delete_time to build the composed primary key, we have to set
+  -- this to `0` instead, since it's not allowed to use a nullable column as part of the primary key.
+  create_time biguint NOT NULL DEFAULT 0,
+  delete_time biguint NOT NULL DEFAULT 0,
+
+  CONSTRAINT pk_sla_lifecycle PRIMARY KEY (id, delete_time)
+);
+
+COMMENT ON COLUMN sla_lifecycle.id IS 'host.id or service.id depending on the checkable type';
```

Comment:
> depending on the checkable type

This is a bit implicit. Regardless of having the same `id` field[^1] for both hosts and services, I would make this explicit by stating that a `NULL` value for `serivce_id` implies a host checkable, et vice versa.


[^1]: `2^(8*20) = 2^160` should be unlikely enough for collisions, unless a crafted SHA-1 attack appears.

Reply:
This is only a suggestion for the column comment as there are already `host_id` and `service_id` columns as well in the table, isn't it?

Regarding collisions: well that's nothing special for this table, it would be the same for collisions between two different hosts for example.

Reply:
> This is only a suggestion for the column comment as there are already `host_id` and `service_id` columns as well in the table, isn't it?

Exactly.

> Regarding collisions: well that's nothing special for this table, it would be the same for collisions between two different hosts for example.

Compared to the other tables, a potential collision between hosts and services would be possible here. But, as already stated, this is a non-issue and I should not have brought it up.

Reply:
Thus, I would rephrase my initial comment here to:

Please make the comment for `id` more explicit regarding the "checkable type" part, which can be derived from a `NULL` value for the `serivce_id`.